samples/va_intel/va_intel_interop.cpp

```diff
@@ -329,24 +349,31 @@ static float run(const char* fn1, const char* fn2, bool doInterop)
     va_status = vaSyncSurface(va::display, surface_id);
     CHECK_VASTATUS(va_status, "vaSyncSurface");
 
-    dumpSurface(va::display, surface_id, fn1, doInterop);
-
     cv::Size size(CLIP_WIDTH,CLIP_HEIGHT);
+    cv::Size size2(IMAGE_WIDTH,IMAGE_HEIGHT);
     cv::UMat u;
+    cv::UMat u2;
 
     cv::va_intel::convertFromVASurface(va::display, surface_id, size, u);
-    cv::blur(u, u, cv::Size(7, 7), cv::Point(-3, -3));
-    cv::va_intel::convertToVASurface(va::display, u, surface_id, size);
+    cv::va_intel::convertFromVASurface(va::display, surface_id2, size2, u2);
+    cv::resize(u, u2, size2);
+    cv::va_intel::convertToVASurface(va::display, u2, surface_id2, size2);
+
+    dumpSurface(va::display, surface_id2, fn1, doInterop);
+    cv::va_intel::convertFromVASurface(va::display, surface_id2, size2, u2);
+    cv::blur(u2, u2, cv::Size(7, 7), cv::Point(-3, -3));
+    cv::va_intel::convertToVASurface(va::display, u2, surface_id2, size2);
     t.start();
-    cv::va_intel::convertFromVASurface(va::display, surface_id, size, u);
-    cv::blur(u, u, cv::Size(7, 7), cv::Point(-3, -3));
-    cv::va_intel::convertToVASurface(va::display, u, surface_id, size);
+    cv::va_intel::convertFromVASurface(va::display, surface_id2, size2, u2);
+    cv::blur(u2, u2, cv::Size(7, 7), cv::Point(-3, -3));
```

Comment:
Changed blur kernel size 7x7 to 3x3; time improved from 7 to 5 ms on 
1920x1080 image.

On 09/18/2015 01:16 PM, krodyush wrote:
>
> In samples/va_intel/va_intel_interop.cpp 
> <https://github.com/Itseez/opencv/pull/5363#discussion_r39841288>:
>
> >      t.start();
> > -    cv::va_intel::convertFromVASurface(va::display, surface_id, size, u);
> > -    cv::blur(u, u, cv::Size(7, 7), cv::Point(-3, -3));
> > -    cv::va_intel::convertToVASurface(va::display, u, surface_id, size);
> > +    cv::va_intel::convertFromVASurface(va::display, surface_id2, size2, u2);
> > +    cv::blur(u2, u2, cv::Size(7, 7), cv::Point(-3, -3));
>
> Is it possible to use faster 3x3 blur or maybe another faster 
> pixel-wise operation to emphasize interoperability benefits?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Itseez/opencv/pull/5363/files#r39841288>.
>